use-cases/typosquat_detection/README.md

```diff
@@ -0,0 +1,448 @@
+* Start the squid parser: `$METRON_HOME/bin/start_parser_topology.sh -z $ZOOKEEPER -s squid`
+
+## Set up Enrichment, Threat Intel and Threat Triage
+
+Now that we have squid parser running, we should create an enrichment to add a field `is_potential_typosquat`
+which determines if the domain is potentially a typosquatted domain.  Furthermore, we should set an alert if 
+it's so and triage those messages.
+
+We can do this by creating `$METRON_HOME/config/zookeeper/enrichments/squid.json` with the following content:
+```
+{
+  "enrichment": {
+    "fieldMap": {
+      "stellar" : {
+        "config" : [
+          "domain_without_tld := DOMAIN_REMOVE_TLD(domain_without_subdomains)",
+          "is_potential_typosquat := BLOOM_EXISTS(OBJECT_GET('/tmp/reference/alexa10k_filter.ser'), domain_without_tld)",
+          "domain_without_tld := null"
+        ]
+      }
+   }
+  ,"fieldToTypeMap": { }
+  },
+  "threatIntel": {
+    "fieldMap": {
+      "stellar" : {
+        "config" : [
+          "is_alert := (exists(is_alert) && is_alert) || is_potential_typosquat"
+        ]
+      }
+
+    },
+    "fieldToTypeMap": { },
+    "triageConfig" : {
+      "riskLevelRules" : [
+        {
+          "name" : "Alexa 10k Typosquat Bloom",
+          "comment" : "Inspect a bloom filter with potentially typosquatted domains from the top Alexa 10k",
+          "rule" : "is_potential_typosquat != null && is_potential_typosquat",
+          "score" : 10,
+          "reason" : "FORMAT('%s is a potential typosquatted domain from the top 10k domains from alexa', domain_without_subdomains)"
+        }
+      ],
+      "aggregator" : "MAX"
+    }
+  }
+}
+
+```
+
+As you can see, following the pattern of enrichments the following are done:
+* A new field `is_potential_typosquat` is created which indicates whether the domain sans TLD and subdomains is a typosquatted domain according to our bloom filter of the top 10k Alexa domains
+* `is_alert` is updated based on the `is_potential_typosquat` field
+* A new threat triage rule is added to give the analyst sufficient context if this alert triggers and a score of 10.
+
+Push the configs via `$METRON_HOME/bin/zk_load_configs.sh -m PUSH -i $METRON_HOME/config/zookeeper -z $ZOOKEEPER`
+
+## Generate Sample Data
+
+We can now use `squidclient` to visit a regular domain and typosquatted domain and send the data to kafka:
+```
+squidclient http://www.github.com
+squidclient http://gituub.com/apache/metron
+cat /var/log/squid/access.log | /usr/hdp/current/kafka-broker/bin/kafka-console-producer.sh --broker-list $BROKERLIST --topic squid
+```
+
+## Investigate via the Alerts UI
+
+We should now have data in our elasticsearch indices, so let's investigate via the alerts UI.   Before we do,
+we have to adjust the mappings for the indices we just created to add the `alert` nested property.  You can
+do that via the following:
+```
+curl -XPUT "http://$ES_HOST/squid*/_mapping/squid_doc" -d '{
```

Comment:
You should just be able to do:
```
curl -XPUT "http://$ES_HOST/squid*/_mapping/squid_doc" -d '
{
        "properties" : {
          "alert" : {
            "type" : "nested"
          }
        }
}
'
```

Adding a mapping should be able to be done without having the whole template, iirc.

Reply:
Also, this whole segment is 2.x format right (e.g. string types)?  If it is necessary in full, it'll have to migrate to 5.x.

Reply:
Very good catch.  I adjusted it and tried it and it worked.  IF you still have vagrant up, would you mind updating and using the new mapping?